Security fix:

off-by-one buffer overflow fix while processing the QUESTION section


$OpenBSD: patch-packet_c,v 1.1.2.1 2009/08/14 01:55:54 william Exp $
--- packet.c.orig	Tue Aug 11 23:22:48 2009
+++ packet.c	Tue Aug 11 23:23:08 2009
@@ -272,8 +272,8 @@ int packet_read_query_section(buffer_type *packet,
 		 * MAXDOMAINLEN ...
 		 */
 		if ((*src & 0xc0) ||
-		    (src + *src + 1 > buffer_end(packet)) || 
-		    (src + *src + 1 > query_name + MAXDOMAINLEN))
+		    (src + *src + 2 > buffer_end(packet)) || 
+		    (src + *src + 2 > query_name + MAXDOMAINLEN))
 		{
 			return 0;
 		}
